src/app/core/entity-components/entity-list/entity-list.component.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild,
+} from "@angular/core";
+import { MatSort } from "@angular/material/sort";
+import { MatPaginator, PageEvent } from "@angular/material/paginator";
+import { Entity } from "../../entity/entity";
+import { MatTableDataSource } from "@angular/material/table";
+import {
+  FilterSelection,
+  FilterSelectionOption,
+} from "../../filter/filter-selection/filter-selection";
+import { User } from "../../user/user";
+import { SessionService } from "../../session/session-service/session.service";
+import { MediaChange, MediaObserver } from "@angular/flex-layout";
+import { ActivatedRoute, Router } from "@angular/router";
+import { EntityMapperService } from "../../entity/entity-mapper.service";
+
+export interface ColumnGroup {
+  name: string;
+  columns: string[];
+}
+
+@Component({
+  selector: "app-entity-list",
+  templateUrl: "./entity-list.component.html",
+  styleUrls: ["./entity-list.component.scss"],
+})
+export class EntityListComponent<T extends Entity>
+  implements OnChanges, AfterViewInit {
+  @Input() entityList: T[] = [];
+  @Input() listConfig: any = {};
+  // This name is used to store the pagination settings in the user entity
+  @Input() componentName: string = "";
+  @Output() elementClick = new EventEmitter<T>();
+  @Output() addNewClick = new EventEmitter();
+
+  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  listName = "";
+  columns: any[] = [];
+  columnGroups: ColumnGroup[] = [];
+  defaultColumnGroup = "";
+  mobileColumnGroup = "";
+  filtersConfig: any[] = [];
+
+  ready = true;
+  columnsToDisplay: any[] = [];
```

Comment:
```suggestion
  columnsToDisplay: string[] = [];
```

array of string column ids, isn't it?